### PR TITLE
[Interop][testing] Use libc++ from target Darwin SDK for clang check …

### PR DESCRIPTION
**Pull request by hyp**
…invocations

### CODE REVIEW

**Comment by hyp** [comment]
@swift-ci please test apple silicon

**Review by edymtt — APPROVED** [review]

File: test/lit.cfg [hunk 1]
```diff
@@ -2297,6 +2265,45 @@ if platform.system() != 'Darwin' or swift_test_mode == 'optimize_none_with_impli
...
+  # Don't warn about any C++98 compatibility issues.
+  '-Wno-c++98-compat-pedantic '
+  # Don't warn about poison system directories in the Linux system paths.
+  '-Wno-poison-system-directories '
+  # For -fmodules-cache-path, when modules are off.
+  '-Wno-unused-command-line-argument '
+  # Don't warn about nullability specifiers in C/C++ code.
+  '-Wno-nullability-extension ')
+
+# Verifies that a C++ file can be compiled without warnings except for some exceptions.
```
Comment (edymtt) [inline test/lit.cfg:2291]: Small nitpick, optional

```suggestion
# Verifies that a C file can be compiled without warnings except for some exceptions.
```

**Comment by hyp** [comment]
@swift-ci please test macOS platform

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test apple silicon

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test apple silicon

**Comment by hyp** [comment]
@swift-ci please test linux platform

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test apple silicon

**Comment by hyp** [comment]
@swift-ci please test linux platform

**Comment by edymtt** [comment]
@swift-ci please test macOS

**Pull request merged**